# import necessary packages

In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
sns.set()

# load the dataset

In [ ]:
url = 'https://raw.githubusercontent.com/jbrownlee/Datasets/master/haberman.csv'
list_cols = ['Age', "Patient's Years", "N_positive_ax", "survival_status"]
cancer_df = pd.read_csv(url, names=list_cols)

In [ ]:
cancer_df.head()

Age  Patient's Years  N_positive_ax  survival_status
0   30               64              1                1
1   30               62              3                1
2   30               65              0                1
3   31               59              2                1
4   31               65              4                1

In [ ]:
cancer_df['survival_status'].value_counts()

1    225
2     81
Name: survival_status, dtype: int64

Bagi kedua data ini menjadi data training dan data test dengan test_size=0.25. (coding)

In [ ]:
X = cancer_df.drop(['survival_status'], axis=1)
y = cancer_df['survival_status']

# spliting data
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.25, random_state=21, stratify=y)

Berapa score rata-rata dari model dengan teknik cross-validation tersebut?

In [ ]:
from sklearn.tree import DecisionTreeClassifier

In [ ]:
from sklearn.model_selection import cross_val_score,cross_validate
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import confusion_matrix, classification_report
model_lr = DecisionTreeClassifier(max_depth=3)
cv_score = cross_val_score(model_lr, X,y, scoring='roc_auc', cv=5)
print("Scores :", cv_score.mean())

Scores : 0.5602532679738562


Prediksi data test dengan model yang telah kalian buat!

In [ ]:
model_lr.fit(X_train, y_train)
y_pred = model_lr.predict(X_test)
acc_decission_tree = round(model_lr.score(X_train, y_train)* 100,2)
print(round(acc_decission_tree, 2), "%")

79.91 %


Bagaimana hasil confusion matrix dari hasil prediksi tersebut?

In [ ]:
confusion_matrix(y_test, y_pred, labels=[1,2])

array([[47, 10],
       [12,  8]])

Bagaimana classification report dari hasil prediksi tersebut?

In [ ]:
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           1       0.80      0.82      0.81        57
           2       0.44      0.40      0.42        20

    accuracy                           0.71        77
   macro avg       0.62      0.61      0.62        77
weighted avg       0.71      0.71      0.71        77



Seberapa baik model anda dalam memprediksi seorang pasien mempunyai status positive? (1)

model berjalan dengan cukup baik dengan hasil precision recall dan f1 score yang menunjukan tingkat keselamatan > 80%

Seberapa baik model anda dalam memprediksi seorang pasien mempunyai status negatif? 

model berjalan dengan cukup baik dengan hasil precision recall dan f1 score yang menunjukan tingkat kematian < 44%